functional/ContentInputStream.java

```diff
@@ -192,7 +192,7 @@ public long skip(long n) throws IOException {
     }
 
     if (available <= 0) {
-      throw new IOException("no more data");
+      throw new EOFException("no more data available");
```

Comment:
`EOFException` is not used anywhere.  Why can't we retain `IOException`?

Reply:
EOF is more appropriate I believe we can keep IOException as well.

Reply:
`EOFException` is subclass of `IOException` i.e. someone catching `IOException` is also valid